### PR TITLE
Dev Guide: Update with debug console for initrd

### PR DESCRIPTION
**Pull request by nitkon**
Update the developer guide with instructions to
attach to the debug console of a initrd rootfs based
VM. 

Fixes:  #502

Signed-off-by: Nitesh Konkar niteshkonkar@in.ibm.com

### CODE REVIEW

File: Developer-Guide.md [hunk 1]
```diff
@@ -646,7 +646,10 @@ the following steps, which assume the use of a rootfs image.
 > `config.sh`
 > file](https://github.com/kata-containers/osbuilder/blob/master/rootfs-builder/clearlinux/config.sh)).
 > For non-systemd based distros, you will need to create an equivalent system
-> service using that distros init system syntax instead.
+> service using that distros init system syntax instead. Another way is to build a non-systemd based
```
Comment (grahamwhaley) [inline Developer-Guide.md]: Thanks for pushing this @nitkon .  I think you've inherited some long sentences etc. in that block already. I've had a stab at reflowing it below (ignore the on-screen format, github rendering of the doc should sort that out once its in a PR).
let me know what you think. /cc @jodh-intel 

---
> **Note:** The debug console instructions below assume a systemd based guest
> O/S image, meaning you must create a rootfs for a distro that supports
> systemd. Currently, all distros supported by
> [osbuilder](https://github.com/kata-containers/osbuilder) support systemd,
> except for Alpine Linux.
>
> To check if an osbuilder distro supports systemd, look for `INIT_PROCESS=systemd` in the `config.sh`
> osbuilder rootfs config file for the distro you wish to build a rootfs for
> (for example see the
> [Clear Linux `config.sh` file](https://github.com/kata-containers/osbuilder/blob/master/rootfs-builder/clearlinux/config.sh)).
>
> For non-systemd based distros, you will need to create an equivalent system
> service using that distros init system syntax instead, or build a 
> distro containing a shell such as `bash(1)`  ( which you will likely need to install additional packages
> in the rootfs to get), and add "agent.debug_console" to kernel parameters in the runtime config file, to tell the Kata agent to launch the console directly.
>
> You can then connect to the virtual machine using the
> [debug console](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#connect-to-the-virtual-machine-using-the-debug-console)
---

**Comment by nitkon** [comment]
@jodh-intel : ptal

File: Developer-Guide.md (hunk 1 above)
Comment (jodh-intel) [reply Developer-Guide.md]: @grahamwhaley - sounds good to me. I do wonder if we can make this even shorter though.

@klynnrif might have thoughts on this.

Comment (klynnrif) [reply Developer-Guide.md]: @jodh-intel @grahamwhaley @nitkon I've taken a shot as well: 

Note: The following debug console instructions assume a systemd-based guest O/S image. This means you must create a rootfs for a distro that supports systemd. Currently, all distros supported by osbuilder support systemd except for Alpine Linux. 

Look for INIT_PROCESS=system in the config.sh osbuilder rootfs config file to verify an osbuilder distro supports systemd for the distro you want to build rootfs for. For an example, see the Clear Linux config.sh file.

For a non-systemd-based distro, create an equivalent system service using that distro’s init system syntax. Alternatively, you can build a distro that contains a shell (e.g. bash(1)). In this circumstance it is likely you need to install additional packages in the rootfs and add “agent.debug_console” to kernel parameters in the runtime config file. This tells the Kata agent to launch the console directly. 

Once these steps are taken you can connect to the virtual machine using the debug console.

Comment (nitkon) [reply Developer-Guide.md]: Any thoughts? @jodh-intel @grahamwhaley

Comment (jodh-intel) [reply Developer-Guide.md]: @nitkon - @klynnrif is the master wordsmith so I vote for her wording!! :smile: 

You'll need to add back in the bold and fixed-font bits from @grahamwhaley's draft though.

Comment (nitkon) [reply Developer-Guide.md]: Updated my PR. ptal...

**Pull request merged**